### PR TITLE
Several API Improvements 

### PR DESCRIPTION
**Pull request by cryptokat**
### Rename API 1.1.0 to API 2.0.0

Considering that we are not going to bump API version for every single release of Semux wallet, it makes more sense to let API have its own versioning numbers following semver.

As the v1.1.0 API introduces a few incompatible changes with v1.0.1 API, it should be named as v2.0.0 instead.

### Improve Naming of API 2.0 Endpoints

Http request method & naming of API 2.0 endpoints are improved following REST semantic web standard.

### Add /compose-raw-transaction & /sign-raw-transaction

to address #766

### CODE REVIEW

**Comment by orogvany** [comment]
This should make it easier to implement the securities for POSTs on #579

File: src/main/java/org/semux/core/Transaction.java [hunk 1]
```diff
@@ -238,6 +238,38 @@ public long getTimestamp() {
...
+        return new Transaction(
+                Network.of(networkId),
+                TransactionType.of(type),
+                to,
+                value,
+                fee,
+                nonce,
+                timestamp,
+                data);
+    }
```
Comment (yulongaion) [inline src/main/java/org/semux/core/Transaction.java]: Can we abstract this decoding logic into an method, shared by the constructor and this method?

Comment (semuxdev) [inline src/main/java/org/semux/core/Transaction.java]: Can we abstract this decoding logic into a method, shared by the constructor and this method?

Comment (cryptokat) [reply src/main/java/org/semux/core/Transaction.java]: fixed by https://github.com/semuxproject/semux/pull/771/commits/c0244233b2eb643a975007ecf302cbf191155c4b

**Pull request merged**